### ISSUE TITLE
UI: Parking Tool

### ISSUE THREAD

**Issue by aubergine10** [krzychu124/Cities-Skylines-Traffic-Manager-President-Edition#47]
## Parking icons

When hovering an icon, associated lane should be highlighted. See #33.

## Car parks / Park and ride

Update to **Traffic** info view:

> By "parking spaces" I mean a place where car can park, regardless of whether that space is currently available or not. The value of '10' is just a guess, based on most growables having fewer than 10 parking spaces.

* Red gradient buildings = cims struggling to find parking when travelling to that building
    * Existing functionality
* White buildings = not enough parking to be considered a car park / no parking issues
* Shades of green = buildings with 10+ parking areas are considered car parks:
    * Light green = Few spaces (>10)
    * ...gradient...
    * Dark green = Loads of parking spaces (>100, like [this](https://steamcommunity.com/sharedfiles/filedetails/?id=485015540) or [this](https://steamcommunity.com/sharedfiles/filedetails/?id=853655964)...)
* Blue buildings = Park and Ride

To designate a building as Park and Ride, user clicks a green-shaded building. That building will then be associated with nearaby PT stations/stops, and will be preferred parking choice - if it's full when driver gets there, try (in order): another blue building, or green building, or roadside parking, or small parking (white buildings).

**Comment by aubergine10** [comment]
## Park and Ride

In the Traffic info view, click a building to select it. It turns blue indicating it's selected. Let's call this building a **hub**:

* A "lane connector" style line then connects mouse pointer to the **hub**.
* User can then click on one or more other buildings to 'connect' them to the **hub**.
    * Let's call these **spokes**.
* The **spokes** turn light blue; clicking them a second time will disconnect them from the **hub**

Once the user is done, they can right click to exit from the currently selected **hub**:

* The **spokes** for that hub go back to their normal colours
* The **hub** remains blue? (not sure about this; what if it is a red "can't find parking" building?)
* User can now select another **hub** to work with.

It's a bit like on lane connector tool where you right-click out of a junction and then you can select a different junction to work with.

If the hub is a station, the spokes can be car parks: Places where cims will prefer to park before they ride at that station. If cims find those car parks full, they'll return to usual 'search nearest' approach.

If the hub is a RICO, or unique, or monument, or park, or park entrance, spokes can be car parks and/or stations.

Essentially user is defining "preferred stations" and "preferred car parks" for a given destination.

**Use cases 1: Observatory**

I have an observatory at the top of a hill, there's not much parking. I click the observatory (hub) and connect it to nearby cable car station (spoke). Now cims going to the observatory will try and use the route associated with the cable car. At the bottom of the hill, I connect a cable car station (hub) to a large car park and a rail station. 

A cim wanting to get to observatory checks if it's a hub: It is! So they want the cable car, which means they want that to either park in the car park or arrive via the train station.

**Use case 2: Train station**

Cims are struggling to find parking for a station (hub) so I connect it to some nearby carparks that might otherwise be overlooked by cims. Now cims doing park and ride use the preferred car parks for that station, leaving smaller car parks free for shoppers or residents.

**Comment by VictorPhilipp** [comment]
What about an automated approach?
1. An algorithm identifies potential hubs (parking buildings placed by user, no RICO).*
2. Then, for every public transport station/stop, find all potential hubs in the vicinity.
3. Check if there is a pedestrian path between the PT station and the potential hub. If **yes**: Mark potential hub as *valid*. If **no**: Mark potential hub as *invalid*
4. During vehicle path-finding, if the path contains a PT station: Select a (random) valid hub of the first PT station in the path and perform another custom Parking AI path-finding to make the car park at the hub.

**Comment by aubergine10** [comment]
Yes, automating good defaults would be desirable. However, as a player, I'd still want to be able to customise it (ie. it becomes a gameplay element).
